middleware/http/wasm/httpwasm.go

```diff
@@ -0,0 +1,162 @@
+	return &middleware{logger: logger}
+}
+
+// GetHandler returns the HTTP handler provided by wasm basic middleware.
+func (m *middleware) GetHandler(metadata dapr.Metadata) (func(next http.Handler) http.Handler, error) {
+	rh, err := m.getHandler(metadata)
+	if err != nil {
+		return nil, err
+	}
+	runtime.SetFinalizer(rh, (*requestHandler).Close)
+	return rh.requestHandler, nil
+}
+
+// getHandler is extracted for unit testing.
+func (m *middleware) getHandler(metadata dapr.Metadata) (*requestHandler, error) {
+	meta, err := m.getMetadata(metadata)
+	if err != nil {
+		return nil, fmt.Errorf("wasm basic: failed to parse metadata: %w", err)
+	}
+
+	var stdout, stderr bytes.Buffer
+	mw, err := wasmnethttp.NewMiddleware(ctx, meta.guest,
+		handler.Logger(m),
+		handler.ModuleConfig(wazero.NewModuleConfig().
+			WithStdout(&stdout).  // reset per request
+			WithStderr(&stderr)), // reset per request
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: meta.PoolSize
+
+	return &requestHandler{mw: mw, logger: m.logger, stdout: &stdout, stderr: &stderr}, nil
+}
+
+// IsEnabled implements the same method as documented on api.Logger.
+func (m *middleware) IsEnabled(api.LogLevel) bool {
+	return true // l.logger.GetOutputLevel()
+}
+
+// Log implements the same method as documented on api.Logger.
+func (m *middleware) Log(_ context.Context, level api.LogLevel, message string) {
+	switch level {
+	case api.LogLevelError:
+		m.logger.Error(message)
+	case api.LogLevelWarn:
+		m.logger.Warn(message)
+	case api.LogLevelInfo:
+		m.logger.Info(message)
+	case api.LogLevelDebug:
+		m.logger.Debug(message)
+	default: // same as api.LogLevelNone
+		return
+	}
+}
+
+func (m *middleware) getMetadata(metadata dapr.Metadata) (*middlewareMetadata, error) {
+	b, err := json.Marshal(metadata.Properties)
+	if err != nil {
+		return nil, err
+	}
+
+	var data middlewareMetadata
+	err = json.Unmarshal(b, &data)
+	if err != nil {
+		return nil, err
+	}
+
+	if data.Path == "" {
+		return nil, errors.New("missing path")
+	}
+
+	data.guest, err = os.ReadFile(data.Path)
+	if err != nil {
+		return nil, fmt.Errorf("error reading path: %w", err)
+	}
+
+	return &data, nil
+}
+
+type requestHandler struct {
+	mw             wasmnethttp.Middleware
+	logger         logger.Logger
+	stdout, stderr *bytes.Buffer
+}
+
+func (rh *requestHandler) requestHandler(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		h := rh.mw.NewHandler(ctx, next)
+		defer func() {
+			rh.stdout.Reset()
+			rh.stderr.Reset()
+		}()
+
+		h.ServeHTTP(w, r)
+
+		if stdout := rh.stdout.String(); len(stdout) > 0 {
+			rh.logger.Debugf("wasm stdout: %s", stdout)
+		}
+		if stderr := rh.stderr.String(); len(stderr) > 0 {
+			rh.logger.Debugf("wasm stderr: %s", stderr)
+		}
+	})
+}
+
+// Close implements io.Closer
+func (rh *requestHandler) Close() error {
+	// TODO: we have to use a finalizer as there's no way in dapr to close middleware, yet.
```

Comment:
I think you can remove the TODO. Although Dapr doesn't close middleware components yet, we will implement it with `Close() error` (= `io.Closer`).